example/tests/integration/test_model_resource_name.py

```diff
@@ -74,10 +93,33 @@ def test_resource_name_precendence(self, client):
         assert (data.get('type') == 'resource_name_from_view'), (
             'resource_name from view incorrect on list')
 
+    def test_model_resource_name_create(self, client):
+        models.Comment.__bases__ += (_PatchedModel,)
+        models.Entry.__bases__ += (_PatchedModel,)
+        response = client.post(reverse("comment-list"),
+                               dump_json(self.create_data),
+                               content_type='application/vnd.api+json')
+
+        assert response.status_code == status.HTTP_201_CREATED
+
+    def test_serializer_resource_name_create(self, client):
+        serializers.CommentSerializer.Meta.resource_name = "renamed_comments"
```

Comment:
Good idea! I replaced all the calls, except the assignment of __bases__, which seems to be impossible (see https://github.com/pytest-dev/pytest/issues/1938)